big_tests/tests/pubsub_tools.erl

```diff
@@ -639,19 +642,35 @@ decode_affiliations(IQResult) ->
 
     [ {exml_query:attr(F, <<"jid">>), exml_query:attr(F, <<"affiliation">>)} || F <- Fields ].
 
-pubsub_node() -> pubsub_node(1).
-pubsub_node(Num) ->
-    Name = <<"node_", (integer_to_binary(Num))/binary, "_",
-             (base64:encode(crypto:strong_rand_bytes(6)))/binary>>,
-    SanitizedName = binary:replace(Name, <<"/">>, <<".">>, [global]),
-    {pubsub_tools:node_addr(), SanitizedName}.
-
 domain() ->
     ct:get_config({hosts, mim, domain}).
 
 node_addr() ->
+    node_addr(<<"pubsub.">>).
+
+node_addr(SubDomain) when is_list(SubDomain) ->
+    node_addr(list_to_binary(SubDomain));
+node_addr(SubDomain) when is_binary(SubDomain) ->
     Domain = domain(),
-    <<"pubsub.", Domain/binary>>.
+    <<SubDomain/binary, Domain/binary>>.
+
+pubsub_node() ->
+    {node_addr(), pubsub_node_name()}.
+
+pubsub_node_with_num(Num) when is_integer(Num) ->
+    Name = <<"node_", (integer_to_binary(Num))/binary, "_",
+             (base64:encode(crypto:strong_rand_bytes(6)))/binary>>,
+    SanitizedName = binary:replace(Name, <<"/">>, <<".">>, [global]),
+    {node_addr(), SanitizedName}.
+
+pubsub_node_with_subdomain(SubDomain) ->
+    {node_addr(SubDomain), pubsub_node_name()}.
+
+pubsub_node_with_num_and_domain(Num, Dom) when is_integer(Num) ->
+    Name = <<"node_", (integer_to_binary(Num))/binary, "_",
+             (base64:encode(crypto:strong_rand_bytes(6)))/binary>>,
+    SanitizedName = binary:replace(Name, <<"/">>, <<".">>, [global]),
```

Comment:
As far as I can tell, this is exactly the same as in `pubsub_node_with_num`. Can this code be used in both functions?